### DISCUSSION TITLE
strongswan 5.9.8 swanctl commands error

### DISCUSSION THREAD

**Discussion by realbitt** [strongswan/strongswan#1467] [Q&A]
new version 5.9.8 show me error when run any swanctl command .

error : 

`plugin 'sqlite': failed to load - sqlite_plugin_create not found and no plugin file available`

is there any solution please ?

**Comment by strongX509** [comment]
The line above is not a fatal error, just a warning that the `sqlite` plugin is not available and cannot be loaded. Actually `swanctl` doesn't need the `sqlite` plugin.

Did you compile strongSwan yourself and did you enabled the `sqlite` plugin some time in the past using `--enable-sqlite` ? Or are you using the standard strongSwan packages from a Linux distribution?

**Reply by realbitt** [reply]
I'm using standard strongSwan packages from a Linux distribution, centos 8 stream
@strongX509

**Reply by tobiasbrunner** [reply]
As Andreas mentioned, this is not a fatal error, so you can just ignore it. But for clarification, it comes from the fact that the _sqlite_ plugin is provided by a separate package for CentOS etc. (`strongswan-sqlite`). However, the config snippet for the plugin is provided by the base package, so there is always an attempt to load the plugin unless it is disabled in the config.

**Reply by jlearman** [reply]
Just FYI, I get the error message (from `swanctl`) even if it's disabled:
```
[root@86a68406345f /]# cat /etc/strongswan/strongswan.d/charon/sqlite.conf
sqlite {
    # Whether to load the plugin. Can also be an integer to increase the
    # priority of this plugin.
    load = no
}
```
Installing strongswan-sqlite makes the error message go away.

**Reply by tobiasbrunner** [reply]
I see. `swanctl` actually doesn't use modular plugin loading or those snippets (they are included in the `charon.plugins` section in the [default strongswan.conf](https://github.com/strongswan/strongswan/blob/7b453ae409ab92cd50568d6554d912cb1e138044/conf/strongswan.conf) file). Instead, it attempts to load all build-enabled [`libstrongswan` plugins](https://docs.strongswan.org/docs/5.9/plugins/plugins.html#_libstrongswan_plugins), which are mostly handling crypto/encoding, but they also include the SQL backends. You'd need to disable `swanctl.plugins.sqlite.load` to prevent `swanctl` from attempting to load the plugin.
